### ISSUE TITLE
Description not announced by Voiceover when following links targeting selects or file inputs

### ISSUE THREAD

**Issue by 36degrees** [alphagov/reported-bugs#3]
**Upstream bug:** https://bugs.webkit.org/show_bug.cgi?id=191749#c0
**Date:** 2018-11-16
**Reported by:** Oliver Byford
**Related to:** https://github.com/alphagov/govuk-frontend/pull/1056#issuecomment-439381905

## Overview

This may well affect other types of inputs, but more severe bugs affecting text inputs, text areas, radio buttons and checkboxes (191746, 191748) means it's impossible to tell.

Given the following markup:

```
  <a href="#select">Link to select</a><br>
  <a href="#file">Link to input type="file"</a>

  <br><br>

  <label for="select">Label for select</label><br>
  <span id="select-description">Description for select</span><br>
  <select id="select" name="select" aria-describedby="select-description">
    <option value="one" selected>Option one</option>
    <option value="two">Option two</option>
    <option value="three">Option three</option>
  </select>

  <br><br>

  <label for="file">Label for file upload</label>
  <span id="file-description">Description for file upload</span>
  <input id="file" name="file" type="file" aria-describedby="file-description">
```

Focus the link to each input type and activate it by double-tapping.

### Expected behaviour:

Voiceover should include the description associated using aria-describedby, as it does when focusing the element by swiping through the document:

"Legend for select, option one, popup button. Description for select. Double-tap to active the picker"

### Actual behaviour:

Voiceover does not include the description as part of the announcement, nor the instruction 'Double-tap to activate the picker':

"Legend for select, option one, popup button."

### Expected behaviour:

Voiceover should include the description associated using aria-describedby, as it does when focusing the element by swiping through the document:

"Label for file upload, no file selected, button. Description for file upload."

### Actual behaviour:

Voiceover does not include the description as part of the announcement, nor the fact that the focussed element is a button:

"Label for file upload, no file selected."

Tested in iOS 12.1 (16B92) on an iPhone X (A1901)

**Comment by querkmachine** [comment]
I've retested this today in iOS 26.4 and got the following announcements when swiping to navigate to elements: 

> Label for select. Description, Description for select. Option one. Popup button. Double tap to activate the picker. 

> Label for file upload. Description, Description for file upload. No file selected. Button. 

And the following when jumping to them by double tapping the links: 

> Label for select. Description, Description for select. Option one. Popup button. 

> Label for file upload. Description, Description for file upload. No file selected. 

The initial issue of not announcing content associated with `aria-describedby` appears to have been resolved in the interim, although it does still omit a level of detail when going via the links. 

As the main aspect of the issue has been resolved, I'm going to close this as having been fixed.
